system/lib/pthread/pthread_create.c

```diff
@@ -61,6 +71,17 @@ int __pthread_create(pthread_t *restrict res, const pthread_attr_t *restrict att
     return EINVAL;
   }
 
+  pthread_t self = __pthread_self();
```

Comment:
This variable is unused now. I think it was probably used in musl to set the thread specific data to `__pthread_tsd_main` when the first thread is created.
https://github.com/emscripten-core/emscripten/blob/4d864df0a57024d667e8db171aab2c3385d04c30/system/lib/libc/musl/src/thread/pthread_create.c#L201